app/routes/endemics/same-herd.js

```diff
@@ -89,7 +105,70 @@ const postHandler = {
     handler: async (request, h) => {
       const { herdSame } = request.payload
       setEndemicsClaim(request, herdSameKey, herdSame)
-      return h.redirect(nextPageUrl)
+
+      const {
+        previousClaims,
+        typeOfReview,
+        dateOfVisit,
+        organisation,
+        typeOfLivestock,
+        latestVetVisitApplication: oldWorldApplication
+      } = getEndemicsClaim(request)
+      const { isReview, isEndemicsFollowUp } = getReviewType(typeOfReview)
+
+      if (herdSame === 'yes') {
+        const prevClaims = previousClaims.filter(claim => claim.data.typeOfLivestock === typeOfLivestock)
+
+        if (isEndemicsFollowUp) {
+          const prevReviewClaim = getReviewWithinLast10Months(
+            dateOfVisit,
+            previousClaims,
+            oldWorldApplication,
+            typeOfLivestock
+          )
+          setEndemicsClaim(request, relevantReviewForEndemicsKey, prevReviewClaim)
+        }
+
+        const errorMessage = canMakeClaim({ prevClaims, typeOfReview, dateOfVisit, organisation, typeOfLivestock, oldWorldApplication })
+        if (errorMessage) {
+          raiseInvalidDataEvent(
+            request, dateOfVisitKey,
+          `Value ${dateOfVisit} is invalid. Error: ${errorMessage}`
+          )
+
+          return h
+            .view(`${endemicsSameHerdException}`, {
+              backLink: pageUrl,
+              errorMessage,
+              ruralPaymentsAgency: config.ruralPaymentsAgency,
+              backToPageText: 'If you entered the wrong date, you\'ll need to go back and enter the correct date.',
+              backToPageMessage: `Enter the date the vet last visited your farm for this ${isReview ? 'review' : 'follow-up'}.`,
+              backToPageLink: dateOfVisitPageUrl
+            })
+            .code(HttpStatus.BAD_REQUEST)
+            .takeover()
+        }
+      } else if (isEndemicsFollowUp) {
```

Comment:
do we need the else here? I think it becomes a little easier to read if we dont have elses unless we need them, and because of the return statement in the IF above, I think we can remove?

Then, it reads down the page:

if X do Y

if A do B

return